### PR TITLE
Toolbar: add basic icon loading ,fix button onclick bug

### PR DESCRIPTION
**Pull request by fanlan1210**

### CODE REVIEW

File: sites/game-client/ui/overlay.mjs [hunk 1]
```diff
@@ -61,8 +76,11 @@ class Overlay {
   */
   onClickToolbarButton() {
     // WIP(fanlan1210)
-    if (this.canHide() && this.position !== undefined) this.hide();
-    else this.show();
+    if (this.canHide() && this.position !== undefined) {
+      this.positionCache = this.position; // store current position state
+      this.hide();
+    }
+    else this.show(this.positionCache); // resore position state
```
Comment (john0312) [inline sites/game-client/ui/overlay.mjs:83]: We might need a default for this.positionCache, or maybe you can add a getDefaultPosition() to Overlay that we'll require the derived class to implement?

File: extensions/chat/client.mjs [hunk 2]
```diff
@@ -36,6 +41,7 @@ class Client {
   async gameStart() {
     this.overlay = new ChatOverlay(this.helper.mainUI);
     this.overlay.show(OverlayPosition.RIGHT);
+    this.helper.mainUI.addToolbarButton(this.overlay);
```
Comment (john0312) [inline extensions/chat/client.mjs:44]: I think this can be done automatically in Overlay and Modal's constructor.

**Review by john0312 — APPROVED** [review]

File: sites/game-client/ui/overlay.mjs [hunk 3]
```diff
@@ -61,8 +79,12 @@ class Overlay {
   */
   onClickToolbarButton() {
     // WIP(fanlan1210)
-    if (this.canHide() && this.position !== undefined) this.hide();
-    else this.show();
+    console.assert(this.positionCache, 'positionCache does not have default value.');
```
Comment (john0312) [inline sites/game-client/ui/overlay.mjs:82]: Optional: I would prefer separating the `positionCache` and the `defaultPosition`.
i.e.
```
if (!this.positionCache) {
  console.assert(this.defaultPosition, 'Derived class should specify this.defaultPosition');
  this.positionCache = this.defaultPosition;
}
```

This way, if something else needs the `defaultPosition`, we'll be able to use it as well.

**Review by john0312 — APPROVED** [review]

File: sites/game-client/ui/overlay.mjs [hunk 4]
```diff
@@ -39,6 +43,20 @@ class Overlay {
...
+  /**
+   * If true, it should set the basic property for toolbarButton,
+   * and mainUI will add a button in toolbar for the overlay.
+   *
+   * *This method can be implemented in the derived class if the derived class
+   * wishes to adjust this default behaviour.*
+   *
+   * @return {Boolean}
+   */
+  hasToolbarbutton() {
```
Comment (john0312) [inline sites/game-client/ui/overlay.mjs:55]: Should be `hasToolbarButton`. (**B**utton, capital B)

Comment (fanlan1210) [reply sites/game-client/ui/overlay.mjs]: resolved in c0e3bc3

**Pull request merged**